nntrainer/tensor/tensor_pool.h

```diff
@@ -340,18 +299,35 @@ class TensorPool {
                         unsigned int offset);
 
 private:
+  /**
+   * @brief Source tensor detailed specification
+   *
+   */
+  struct SourceDetails {
+    unsigned int token;                   /**< memory token */
+    TensorLifespan lifespan;              /**< life span of the tensor */
+    std::vector<unsigned int> exec_order; /**< exec order */
+    std::vector<unsigned int>
+      dependents; /**< list of dependents to the source */
+  };
+
+  /**
+   * @brief Dependent tensor detaild specification
+   *
+   */
+  struct DependentDetails {
+    unsigned int parent_idx; /**< index to the parent */
+    unsigned int offset;     /**< elementwise offset */
+  };
+
   /**
    * @brief Spec for storing each request of tensor from tensor pool
    * @todo move tensor initialization from tensor class to RequestSpec
    */
   struct RequestSpec {
-    std::unique_ptr<Tensor> tensor;       /**< tensor object itself */
-    std::vector<unsigned int> exec_order; /**< tensor exec order list */
-    TensorLifespan lifespan;              /**< tensor lifespan */
-    unsigned int token; /**< tensor memory token or index to source spec */
-    unsigned int offset =
-      0; /**< offset in element from the primary source spec */
-    bool dependent = false; /**< if dependent on another tensor for memory */
+    std::unique_ptr<Tensor> tensor; /**< tensor object itself */
```

Comment:
In my opinion, this could be changed with TensorSpec with Details ( Dependent, Source( I'm still not sure if it is source ) ). But if the user ask requestTensor.. then the tensor pool gives TensorSpec which is created. So.. I thinks.. TensorSpec might be right term.. and for the later usages, we might have to change the TensorSpect like execution order or others.. so.. I guess TensorSpec is better.. One thing bothering me is that we have same term in Context.

Reply:
Yes, I can't agree more. We can have a better name for this structure. It is more of a TensorSpec but we can't use TensorSpec as name clashes, if anyone throws me a suggestion, I'll make a change :)